Src/WindowsServer/WindowsServer.Shared/Implementation/AssemblyLoader.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Reflection;
+    using System.Text;
+
+    // AssemblyLoader contains logic to load Microsoft.WindowsAzure.ServiceRuntime assembly and read properties using reflection
+    // Inherits MarshalByRefObject so that methods of this class can be executed remotely in separate AppDomain.
+    internal class AssemblyLoader : MarshalByRefObject
+    {
+        public string AssemblyName;
+        public string Culture;
+        public string PublicKeyToken;
+        public string[] VersionsToAttempt;
+
+        public bool ReadAndPopulateContextInformation(ref string roleName, ref string roleInstanceId)
+        {
+            Assembly loadedAssembly = null;
+            try
+            {
+                // As this is executed inside a separate AppDomain, it is safe to load assemblies here without interfering with user code.                
+                loadedAssembly = this.AttemptToLoadAssembly(this.AssemblyName, this.Culture, this.PublicKeyToken, this.VersionsToAttempt);
+                if (loadedAssembly != null)
```

Comment:
Instead of writing the logic of 'what to do with the loaded assembly' here, it could be accepted as a function parameter, so as to improve testability...